### PR TITLE
Add interface for Transaction Store

### PR DESCRIPTION
**Pull request by cw9**
@xichen2020 @robskillington @prateek

### CODE REVIEW

File: kv/types.go [hunk 1]
```diff
@@ -105,3 +105,66 @@ type Store interface {
...
+	// SetType sets the type of the operation
+	SetType(ot OpType) Op
+
+	// Key returns the key used in the operation
+	Key() string
+	// SetKey sets the key in the operation
+	SetKey(key string) Op
+
+	// Value returns the value to be set, if the op type is Set
+	Value() proto.Message
```
Comment (xichen2020) [inline kv/types.go]: Talked offline, let's move this outside the `Op` interface and into perhaps a type-specific interface (e.g., `ValueOp` and `NoValueOp`).

File: kv/store.go [hunk 2]
```diff
@@ -20,7 +20,10 @@
 
 package kv
 
-import xwatch "github.com/m3db/m3x/watch"
+import (
+	"github.com/gogo/protobuf/proto"
```
Comment (xichen2020) [inline kv/store.go]: This is not the correct import, you want `"github.com/golang/protobuf/proto"`

File: kv/store.go [hunk 3]
```diff
@@ -88,3 +91,44 @@ func valueFromWatch(value interface{}) Value {
...
+}
+
+type opResponse struct {
+	opBase
+
+	value interface{}
+}
+
+// NewOpResponse returns a OpResponse
+func NewOpResponse(key string, value interface{}) OpResponse {
```
Comment (xichen2020) [inline kv/store.go]: Does this ctor need to be exported?

File: kv/store.go [hunk 4]
```diff
@@ -20,7 +20,10 @@
 
 package kv
```
Comment (xichen2020) [inline kv/store.go]: Might wanna pull op related stuff into `op.go`

Comment (cw9) [inline kv/store.go]: will do

File: kv/store.go (hunk 3 above)
Comment (cw9) [inline kv/store.go]: yeah probably private is enough

**Pull request merged**